### PR TITLE
CC3D: Rename from coptercontrol, remove legacy CC stuff

### LINKED ISSUE #265: Rename coptercontrol -> CC3D
- Will be clearer for users that it works on CC3D
- Will be clearer for users that it doesn't work on original CC

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #265
- Removes ADXL345
- Removes PFC8591
- Removes ADC driver from CC gyros
- Allows saving settings to flash on CC3D without expert mode enabled
- Numerous small tweaks

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/503)

### CODE REVIEW

File: flight/targets/cc3d/fw/pios_board.c [hunk 1]
```diff
@@ -107,14 +107,8 @@ void PIOS_Board_Init(void) {
 	/* Delay system */
 	PIOS_DELAY_Init();
 
-	const struct pios_board_info * bdinfo = &pios_board_info_blob;
```
Comment (mlyle) [inline flight/targets/cc3d/fw/pios_board.c:110]: Are you sure we should remove this?  It seems nice to refuse to start on original CopterControl.

**Comment by mlyle** [comment]
Looks good to me other than that one thing.

File: flight/targets/cc3d/fw/pios_board.c (hunk 1 above)
Comment (tracernz) [inline flight/targets/cc3d/fw/pios_board.c:110]: Not a bad point.

**Comment by tracernz** [comment]
Will also need to rename the board element in the connection diagram. That would conflict with #546 so maybe a good idea to merge it first.

**Comment by mlyle** [comment]
I tried this and it's good.  Ready for merge after rebase.

**Comment by mlyle** [comment]
I noticed one minor thing on windows:  if you install a release over an old one, the old firmware files are left there.  fw_coptercontrol.tlfw is very tempting to click on and results in very confusing outcomes.

**Comment by tracernz** [comment]
I'm guessing we can set the installer up to clean the firmware directory if it's upgrading?

**Comment by mlyle** [comment]
Yes, that would be sane.   http://nsis.sourceforge.net/Reference/RMDir

**Comment by tracernz** [comment]
Rebased, just the removal of UAVObjects.inc causing the conflict.

**Comment by tracernz** [comment]
Blocked on #546 + I need to make the change to the connection diagram once 546 is in.

**Comment by tracernz** [comment]
Fixed now, even tested the connection diagram to make sure.

File: flight/targets/cc3d/fw/pios_board.c (hunk 1 above)
Comment (tracernz) [inline flight/targets/cc3d/fw/pios_board.c:110]: This code does nothing though? There's just a comment saying what it should do.

Comment (mlyle) [inline flight/targets/cc3d/fw/pios_board.c:110]: Haha. OK, I'm awesome.

**Pull request merged**